hypothesis-python/src/hypothesis/extra/_patching.py

```diff
@@ -141,20 +141,38 @@ def leave_FunctionDef(self, _, updated_node):
 
 
 def get_patch_for(func, failing_examples, *, strip_via=()):
-    # Skip this if we're unable to find the location or source of this function.
+    # Skip this if we're unable to find the location of this function.
     try:
         module = sys.modules[func.__module__]
-        fname = Path(module.__file__).relative_to(Path.cwd())
-        before = inspect.getsource(func)
+        file_path = Path(module.__file__)
     except Exception:
         return None
 
+    fname = (
+        file_path.relative_to(Path.cwd())
+        if file_path.is_relative_to(Path.cwd())
+        else file_path
+    )
+    patch = _get_patch_for(
+        func, failing_examples, strip_via=strip_via, namespace=module.__dict__
+    )
+    if patch is None:
+        return None
+
+    (before, after) = patch
+    return (str(fname), before, after)
+
+
+# split out for easier testing of patches in hypofuzz, where the function to
+# apply the patch to may not be loaded in sys.modules.
+def _get_patch_for(func, failing_examples, *, strip_via=(), namespace):
+    try:
+        before = inspect.getsource(func)
+    except Exception:  # pragma: no cover
+        return None
+
```

Comment:
I *probably* could make hypofuzz tests work with the automatic module lookup currently in master, but splitting this seems reasonable for hypothesis' own patching tests as well, and as a general line of separation (module <-> namespace)